src/pseudopeople/noise_functions.py

```diff
@@ -517,7 +519,11 @@ def write_wrong_digits(
     rng = np.random.default_rng(
         get_hash(f"{randomness_stream.seed}_{column_name}_write_wrong_digits")
     )
-    column = column.astype(str)
+    if column_name in INT_TO_STRING_COLUMNS:
+        column = to_string_as_integer(column)
+    else:
+        column = column.astype(str)
```

Comment:
This block occurs a fair amount throughout the codebase. Might be worth extracting to a method.